### ISSUE TITLE
[BUG]: The network icon shows as unavailable even though it is available

### ISSUE THREAD

**Issue by mathmakgakpak** [undergroundwires/privacy.sexy#216]
### Description
The network shows as unavailable even though it is available (I can browse the internet).
I don't know which one script caused it.
I briefly searched (wi-fi, network, icon, internet) open and closed issues but I couldn't find anything related to scripts I chose.
**What's sad that it only happens on my local machine and it works just fine on virtual box machine.**

### OS
Windows 22H2 19045.2728

### Reproduction steps
1. Run the script
2. Restart PC
3. The network shows as unavailable even though it is available
4. You still can browse the internet

### Scripts
https://pastebin.com/a5nCqYku

### Screenshots
![obraz](https://user-images.githubusercontent.com/44949126/229357906-71b46a32-0d2a-4ad3-9c59-082a565fdf8e.png)

**Comment by mathmakgakpak** [comment]
well I can't reproduce it on other computers so I guess I was just unlucky

**Comment by mshpp** [comment]
I would have to reopen this issue, this persists in my case and there's no solution I have found yet. Does anyone have any ideas?

**Comment by undergroundwires** [comment]
Thank you for sharing insights @mshpp, did `Disable active probing to Microsoft NCSI server` resolve the issue for you? Or do you mean something else as NCSI service? I will update the docs, and adjust recommendations based on your feedback.

Was this this script in particular, causing the issue:

```batchfile
:: ----------------------------------------------------------
:: -Disable active probing to Microsoft NCSI server (revert)-
:: ----------------------------------------------------------
echo --- Disable active probing to Microsoft NCSI server (revert)
reg add "HKLM\SYSTEM\CurrentControlSet\Services\NlaSvc\Parameters\Internet" /v "EnableActiveProbing" /t REG_DWORD /d "1" /f
:: ----------------------------------------------------------
```

**Comment by mshpp** [comment]
Correct, it was the "Disable active probing ..." script that caused the issue. Moreover, it seems that the script does need to be disabled/not applied (thus, the probing should be *enabled*) for the WiFi signal indicator to work correctly. Otherwise it works unreliably.

I may be mistaken, but my desktop PC, which is connected through an Ethernet cable, has also had the "Strict" template applied, which includes the "Disable active probing ..." script. However, the issue does not seem to manifest when the connection is wired. Indeed, the normal "monitor with a cable" icon is being shown.

-------- Ursprüngliche Nachricht --------
Am 15.04.24 12:52 um undergroundwires schrieb :

**Comment by undergroundwires** [comment]
Thank you, I will update the dosc with your findings and remove this script from `Strict` so it will no longer be recommended.

I found [this article](https://web.archive.org/web/20121226172641/http://blogs.technet.com/b/networking/archive/2012/12/20/the-network-connection-status-icon.aspx) shared once by Microsoft.

It contacts `www.msftncsi.com` frequently, which still is a privacy issue:

> IIS logs are stored on the server at www.msftncsi.com. These logs contain the time of each access and the IP address recorded for that access. These IP addresses are not used to identify users, and in many cases, they are the address of a network address translation (NAT) computer or proxy server, not a specific client behind that NAT computer or proxy server.
> [Microsoft](https://learn.microsoft.com/en-us/previous-versions/windows/it-pro/windows-vista/cc766017(v=ws.10)?redirectedfrom=MSDN)

I see that it may even break [captive portals](https://learn.microsoft.com/en-us/windows-server/networking/ncsi/ncsi-overview#captive-portals).

In future, to both reach privacy and functionality, the safer way would be to change the NCSI server to somewhere audited and "trusted".

**Comment by 3c7b1b91-0c9d-40a8-88bf-6e3c7f28b3e5** [comment]
Trying to run all Disable Connectivity Checks on Revert mode gives me this error and the icon is broken, as a consequence I can't login on MS Teams anymore.

--- Disable active connectivity tests (breaks internet connection status, captive portals) (revert)
The operation completed successfully.
The operation completed successfully.                                                                                   Note: For the changes to fully take effect, please restart your computer.                                               --- Disable passive connectivity tests (breaks internet connection status) (revert)                                     The operation completed successfully.                                                                                   The operation completed successfully.
--- Remove "Network Connectivity Status Indicator (NCSI)" app (breaks internet connection status icon) (revert)
Removing registry key at "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Appx\AppxAllUserStore\EndOfLife\S-1-5-21-4218298140-1662697426-2370780766-1002\NcsiUwpApp_8wekyb3d8bbwe".
Skipping, no action needed, registry key "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Appx\AppxAllUserStore\EndOfLife\S-1-5-21-4218298140-1662697426-2370780766-1002\NcsiUwpApp_8wekyb3d8bbwe" does not exist.
Starting the installation process for "NcsiUwpApp"...
Checking if "NcsiUwpApp" is installed on another user profile...
Found package "NcsiUwpApp_1000.25128.1000.0_neutral_neutral_8wekyb3d8bbwe".
Manifest file not found for "NcsiUwpApp" on another user profile: "C:\Windows\SystemApps\NcsiUwpApp_8wekyb3d8bbwe\AppxManifest.xml".
Trying to install "NcsiUwpApp" using its package family name: "NcsiUwpApp_8wekyb3d8bbwe" from system installation...
WARNING: Error installing using package family name: Deployment failed with HRESULT: 0x80073CF0, Package could not be
opened.

error 0x80070002: Opening file from location: AppxManifest.xml failed with error: The system cannot find the file
specified.
.

NOTE: For additional information, look for [ActivityId] 456b2e87-67e8-0006-b7dd-6b45e867dc01 in the Event Log or use
the command line Get-AppPackageLog -ActivityID 456b2e87-67e8-0006-b7dd-6b45e867dc01
Unable to reinstall the requested package (NcsiUwpApp). It appears to no longer be included in this version of
Windows. You may search for it or an alternative in the Microsoft Store or consider using an earlier version of
Windows where this package was originally provided.
At line:1 char:2112
+ ... ssage)"; }; throw "Unable to reinstall the requested package ($packag ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Unable to reins...nally provided.:String) [], RuntimeException
    + FullyQualifiedErrorId : Unable to reinstall the requested package (NcsiUwpApp). It appears to no longer be inclu
   ded in this version of Windows. You may search for it or an alternative in the Microsoft Store or consider using a
  n earlier version of Windows where this package was originally provided.

Removing registry key at "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Appx\AppxAllUserStore\Deprovisioned\NcsiUwpApp_8wekyb3d8bbwe".
Skipping, no action needed, registry key "HKLM:\SOFTWARE\Microsoft\Windows\CurrentVersion\Appx\AppxAllUserStore\Deprovisioned\NcsiUwpApp_8wekyb3d8bbwe" does not exist.
--- Block Microsoft connectivity check hosts (breaks internet connection status, captive portals) (revert)
Processing removal for IPv4 entry.
Successfully removed the entry.
Processing removal for IPv6 entry.
Successfully removed the entry.
Processing removal for IPv4 entry.
Successfully removed the entry.
Processing removal for IPv6 entry.
Successfully removed the entry.
Processing removal for IPv4 entry.
Successfully removed the entry.
Processing removal for IPv6 entry.
Successfully removed the entry.
Processing removal for IPv4 entry.
Successfully removed the entry.
Processing removal for IPv6 entry.
Successfully removed the entry.
Processing removal for IPv4 entry.
Successfully removed the entry.
Processing removal for IPv6 entry.
Successfully removed the entry.
Processing removal for IPv4 entry.
Successfully removed the entry.
Processing removal for IPv6 entry.
Successfully removed the entry.
--- Disable "Network Location Awareness (NLA)" service (breaks auto-reconnect, connectivity status, network identification) (revert)
Skipping: Windows (10.0.26200) is above maximum 10.0.19045 (Windows10-MostRecent)
--- Disable "Network List Service (NLS)" service (breaks connectivity status, network identification, network connection icon, connectivity with some Microsoft apps) (revert)
Skipping: Windows (10.0.26200) is above maximum 10.0.19045 (Windows10-MostRecent)

It seems there was an error there.

How can I revert it?

**Comment by ryder203** [comment]
I got the same problem; I ran privacy sexy and then I got confused some days later because my network icon says offline, but I'm online. I want this to function normally. Any fix? Thanks in advance.
